### PR TITLE
Handle module dependencies using composer

### PR DESCRIPTION
**Pull request by Guite**
| Q                 | A
| ----------------- | ---
| Bug fix?          | no
| New feature?      | yes
| BC breaks?        | no
| Deprecations?     | no
| Fixed tickets     | #3388 
| Refs tickets      | -
| License           | MIT
| Changelog updated | yes

This utilises the Composer Merge Plugin for initial composer executions.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -131,31 +63,25 @@
...
-            "type": "vcs",
-            "url": "https://github.com/corejavascript/typeahead.js"
+        "merge-plugin": {
+            "include": [
+                "src/lib/Zikula/Bundle/*/composer.json",
+                "src/system/*/composer.json",
+                "src/modules/*/composer.json",
+                "src/modules/*/*/composer.json",
+                "src/themes/*/composer.json",
+                "src/themes/*/*/composer.json"
```
Comment (craigh) [inline composer.json]: isn't that redundant? doesn't `*` support multi-level?

Comment (Guite) [inline composer.json]: no it does not

Comment (craigh) [inline composer.json]: that could be a problem because there is no guarantee of the composer.json placement level

**Comment by craigh** [comment]
I wasn't clear this was ready. if so (and you know it works) then merge it

**Pull request merged**